src/App.js

```diff
@@ -16,60 +16,92 @@ import apiEndpoints from "./config";
 import { cleanJSON } from "./utils";
 
 const App = () => {
-  const [columns, setColumns] = useState([]);
+  const searchParams = new URLSearchParams(window.location.search);
+
+  const [path, setPath] = useState(searchParams.get("path"));
+  const [columns, setColumns] = useState([
+    { items: [], selected: "published" },
+    ...(path?.split(".").map((element) => ({ items: [], selected: element })) ||
+      []),
+  ]);
   const [dataView, setDataView] = useState("");
   const [apiEndpoint, setApiEndpoint] = useState(apiEndpoints[0].value);
 
   useEffect(() => {
-    fetchChildren(apiEndpoint, "published").then((data) =>
-      setColumns([
-        {
-          items: data.map((name) => ({ name, isSelected: false })),
-          selected: null,
-        },
-      ])
+    const columnPaths = columns.map((_, idx) =>
+      columns
+        .slice(0, idx + 1)
```

Comment:
You seem to be using `columns` inside this effect. Perhaps it should also be part of the dependency array? Any potential issues while syncing up the data?

Reply:
since the columns are being updated inside the array, adding it to the dependency array would result in an infinite loop.
i realize this isnt prefereable but i couldnt think of a better approach. any suggestion is welcome